### DISCUSSION TITLE
The Project itself

### DISCUSSION THREAD

**Discussion by GaborElAyoubi** [flameshot-org/flameshot#3727] [General]
Hello everyone 

I've been using Flameshot for about half a year and was grateful for it until now, but it seems like the project is dead. In the meantime, I have discovered some bugs that severely affect the further work with it and I will probably look for another tool.

Some questions have crossed my mind that might also apply to Github projects in general.

-Why are there so many open pull requests? 

-Is there a switch to Apple Silicon or what does the future of Flameshot look like? 

I'm just wondering how this works and what happens to github projects like Flameshot

**Comment by da411d** [comment]
Same question. Have some errors with multiscreen setup and different DPI, still no fixes =(

**Comment by FelixJochems** [comment]
For clarity: I'm currently working on a re-implementation of flameshot targeted at linux with wayland compositors. I am NOT officially a part of the flameshot project in any way. Just really invested into it and have been working with the codebase for awhile now.

Having said that, the project is massive and there are quite a lot of CMAKE switches specific for apple, windows and linux. The daemonization of the the flameshot program for example has a lot of OS specific things going on. This can result in something being fixed for 1 specific version of an OS but needing to be tested for all other versions, massively slowing down merges.

There also has been an ongoing effort to have plugin support so some things might not get merged on purpose since they fit better as a future plugin.

The project is currently 'stuck' on QT 5.2? Apple silicon has been added in QT 6.2 (think there are some workarounds to get it working in QT5) so would require conversion and even though there are multiple backward compatibility add-ons for the newer QT versions, that code has to be converted at some point. Yet another refactor that requires knowing about all the interacting systems within flameshot.

QT6 has started adding more wayland native elements, which is also the main reason for me starting a fork. Another aspect that, I personally believe, doesn't help is that the ideas for donations/financial support never went anywhere.

Flameshot will hang on for a long time since most breaking changes happen on high DPI, Wayland display servers, Apple silicon or a combination of those. However frustrating it might feel, it's probably still a minority of the entire userbase.

**Comment by mmahmoudian** [comment]
> but it seems like the project is dead. In the meantime

Ni it is definitely not dead. We are glad that you are using it for half a year, but we are using it for 6 years and we are still here.

> I have discovered some bugs that severely affect the further work with it

Users are encouraged to report vugs. That's the responsible thing to do, nit writing posts without explaining the issue.

> will probably look for another tool.

Surely there numerous projects out there. Enjoy the FLOSS world.

> Why are there so many open pull requests?

Because reciewing PRs needs time and should be done carefully to make sure users will not get any malicious code on their computer. And comments like your unfortunately have burnt out multiple of out main developers!

> Is there a switch to Apple Silicon or what does the future of Flameshot look like?

I don't understand your question. How these two are even connected??

> I'm just wondering how this works and what happens to github projects like Flameshot

They stay as active projects as long as there is a community behind it.

**Reply by mmahmoudian** [reply]
@FelixJochems we are actively looking for bew team members. As you correctly said, the project is large and also needs some upgrades. Wayland and Qt6 are things we should do soober or later. If you are interested, okease drop in the matrix group as well.

If you have time and are willing to help, reviewing some of the PRs can be super helpful. The more eyes that are on the codebase, the better and safer the project will be.

**Comment by GaborElAyoubi** [comment]
@mmahmoudian  I didn’t mean to sound negative in my previous comment, I was just genuinely curious. With over 600 open issues and around 70 open pull requests, I wanted to get a sense of where the project stands. I appreciate all the hard work you’ve put in and just wanted to understand if there’s still active development happening.

Thanks for your efforts and dedication to this project!
